### ISSUE TITLE
Stop using EventEmitter

### ISSUE THREAD

**Issue by jrieken** [Microsoft/vscode#38417]
We have two eventing solutions: The old one and not yet fully adopted one... For real, there are 1044 references to the *new* `Event` but still 66 to the *old* `EventEmitter`. That means we are almost done and finish this. 

This is the output from my [reference++](https://marketplace.visualstudio.com/items?itemName=jrieken.references-plusplus) extension, assigning people based on that list

```
Found 2 references

file:///Users/jrieken/Code/vscode/src/vs/base/common/eventEmitter.ts
  194   * Same as EventEmitter, but guarantees events are delivered in order to each listener
  195   */
  196: export class OrderGuaranteeEventEmitter extends EventEmitter {
  34: export class EventEmitter implements IEventEmitter {
  35
  36  	protected _listeners: IListenersMap;
```

**Comment by joaomoreno** [comment]
YES

**Comment by isidorn** [comment]
I will do the `Sash` and the `StartDebugActionItem`

**Comment by sandy081** [comment]
Removed it in `testConfigurationService` and `workbenchTestServices`

**Comment by joaomoreno** [comment]
@jrieken  It would be great to convert this to a checkbox list.

**Comment by isidorn** [comment]
Done. The removal from the ActoinBar ripled through to a lot of files so it might be worth to regenerate that list.

**Comment by rebornix** [comment]
EventEmitter removed from editor area (FindState and ReferenceModel).

**Comment by rebornix** [comment]
I reran the scripts and found that editor modelServiceImpl https://github.com/Microsoft/vscode/blob/master/src/vs/editor/common/services/modelServiceImpl.ts#L10 and viewModelImpl are still relying on EventEmitter.

**Comment by alexandrudima** [comment]
The new event emitter infra does not support the following features:
1. **in order guaranteed event delivery**. i.e.:
   * if an object `A` produces event `e1`.
   * and there are two listeners `listener1` and `listener2`
   * if `listener1` decides to interact again with object `A` which leads to the producing of event `e2`.
   * for certain cases (e.g. buffer and content changes) it is critical that `listener2` first receives `e1` and only then `e2`.
    * as sad as it sounds, I am not making this stuff up, people make edits to models in the change listener, or change decorations in the decorations change listener.

2. `addBulkListener` - I think I can rewrite my code to no longer use it; not 100% sure yet.

**Comment by jrieken** [comment]
@alexandrudima Yeah, did think about in-order delivery yet but I believe it's a good property in general. I have pushed e133265 which optimistically enables that for all `Emitter`s. Tests et all are happy so let's take the risk... If we see issues with it, we can make it in option

**Comment by joaomoreno** [comment]
@jrieken I've converted everything to checkboxes...

**Comment by dbaeumer** [comment]
Done for task specific code.

**Comment by joaomoreno** [comment]
Just did the tree: https://github.com/Microsoft/vscode/commit/6c3c74ab21e463ffc8804d481490df71028beb09 This was an insane refactoring. #GoodLuckDoingThatWithJavaScript #TypeScriptForever

**Comment by jrieken** [comment]
Update the comment with what's left. @joaomoreno there is still a little in for you... Also unsure who owns `contextview.ts`?

**Comment by isidorn** [comment]
I will do the contextview

**Comment by joaomoreno** [comment]
Thanks @isidorn!

**Comment by joaomoreno** [comment]
Did the rest of tree and button. A few todos for [@isidorn](https://github.com/Microsoft/vscode/commit/7f673ae5fdc3dcd048cbd675ee20c948659e303e#diff-8f1d139c14ac71c3411b867e0071fd98R60) and [@jrieken](https://github.com/Microsoft/vscode/commit/7f673ae5fdc3dcd048cbd675ee20c948659e303e#diff-8b65f82e726aab72c24dc88b04ed6308R216). It would be great to get a lint error when we forget to do something with the disposable...

**Comment by joaomoreno** [comment]
Did one last push on the stuff I knew how to. Found quite a few places where we don't collect the disposables from events... fixed what I could and left `TODO@[name]` markers behind otherwise.

I believe only `TextModel` is left... 💃 

Thanks for the push @jrieken!

**Comment by isidorn** [comment]
@joaomoreno thanks, the only one I found is in empty view and fixed via 738d907bda

**Comment by joaomoreno** [comment]
![witch](https://media.giphy.com/media/BMzxYSahXu7YY/giphy.gif)
